### PR TITLE
Add swift docs

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

This PR seeks to add docs for the library that  Maail has kindly put together.

## What is the current behavior?

No swift docs

## What is the new behavior?

Swift docs v1 

## Additional context

Reference document is on Notion, will populate the reference gradually.
- SVG Icons are placeholders and will need to be replaced nearing the completion of this PR
- There's still parts of the frontend which need to completed by referencing the [C# PR](https://github.com/supabase/supabase/pull/11239)

### CODE REVIEW

**Comment by J0** [comment]
CleanShot 2023-04-01 at 00 09 42@2x
Think there are a few sections which we still need to fill with data/response

File: spec/supabase_swift_v1.yml [hunk 1]
```diff
@@ -0,0 +1,2687 @@
...
+                .from("countries")
+                .select("name")
+                .or(filters: "id.gt.3,and(id.eq.1,name.eq.Afghanistan)")
+                .execute()
+          ```
+
+  - id: not
+    title: not()
+    description: |
+      Finds all rows which doesn't satisfy the filter.
```
Comment (saltcod) [inline spec/supabase_swift_v1.yml:547]:
```suggestion
      Finds all rows that don't satisfy the filter.
```

File: spec/supabase_swift_v1.yml [hunk 2]
```diff
@@ -0,0 +1,2687 @@
...
+      - id: verify-signup-one-time-password(otp)
+        name: Verify Signup One-Time Password (OTP)
+        isSpotlight: false
+        code: |
+          ```js
+             try await supabase.auth.verifyOTP(email: "example@example-email.com",
+                                  token: "123456",
+                                  type: .signup)
+          ```
+  - id: get-session
```
Comment (saltcod) [inline spec/supabase_swift_v1.yml:2163]: Confirming these aren't missing from this somehow, but are just going to come later? 
(supabase-js on the right here)

CleanShot 2023-04-03 at 09 27 10@2x

File: spec/storage_v0_openapi.json [hunk 3]
```diff
@@ -708,6 +730,118 @@
...
+            "content": {
+              "application/json": {
+                "schema": { "description": "Error response", "$ref": "#/components/schemas/def-1" }
+              }
+            }
+          }
+        }
+      },
+      "put": {
+        "summary": "Uploads an object via a presigned URL",
```
Comment (saltcod) [inline spec/storage_v0_openapi.json:790]: curious about the params here in supabase-js.   Are these not relevant for swift?  Or just not handled yet?   Want to make sure I understand! 


CleanShot 2023-04-03 at 09 30 42@2x

File: spec/supabase_swift_v1.yml (hunk 2 above)
Comment (maail) [reply spec/supabase_swift_v1.yml:2163]: I might have missed it. Let me take a look and update.

File: spec/common-client-libs-sections.json [hunk 4]
```diff
@@ -855,10 +858,7 @@
   },
   {
     "title": "Misc",
-    "excludes": [
-      "reference_dart_v1",
-      "reference_dart_v0"
-    ],
+    "excludes": ["reference_dart_v1", "reference_dart_v0"],
```
Comment (thorwebdev) [inline spec/common-client-libs-sections.json:861]: I think we need to add `reference_swift_v1` here as it's currently a dead item in the nav: 
image

**Review by thorwebdev — COMMENTED** [review]
Wah, this looks incredible, thanks so much. Just a small nit regarding the dead MISC menu item, otherwise LGTM 👍

File: spec/supabase_swift_v1.yml (hunk 2 above)
Comment (maail) [reply spec/supabase_swift_v1.yml:2163]: Hmm code examples should be there. I'm seeing it on dark mode @J0 how about for you?

I'm updating some code there though which isn't correct.

File: spec/storage_v0_openapi.json (hunk 3 above)
Comment (maail) [reply spec/storage_v0_openapi.json:790]: @saltcod got what you meant. I was working of Flutter V1 and JS V1 docs I think so the parameter details weren't there in those. I'll add them in the next version.

File: spec/common-client-libs-sections.json (hunk 4 above)
Comment (maail) [reply spec/common-client-libs-sections.json:861]: @J0 would you be able to help with this one?

**Review by thorwebdev — APPROVED** [review]
LGTM 💚

**Pull request merged**